kore/src/Kore/Step/Rule.hs

```diff
@@ -777,3 +780,48 @@ substitute subst rulePattern' =
         }
   where
     RulePattern { left, right, requires, ensures } = rulePattern'
+
+{- | Context in which a step is taken
+-}
+data StepContext
+  = SimplificationContext
+  | DefinitionContext
+  | RewriteContext
+  | OnePathContext
+  | AllPathContext
+  deriving (Show, Eq)
+
+
+{- | Check if the term and the 'left' fields of the rules are function-like
+
+-}
+
+checkFunctionLike
+    ::  (Debug variable
+        , FreshVariable variable
+        , SortedVariable variable
+        , Show variable
+        , Unparse variable
+        , Debug variable'
+        , FreshVariable variable'
+        , SortedVariable variable'
+        , Show variable'
+        , Unparse variable')
+    => [RulePattern variable']
```

Comment:
There is nothing here which indicates that this should be the list of _applied_ rules, yet that seems to be critical.